src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -154,6 +157,24 @@ protected JestElasticsearchClient(Map<String, String> props, JestClientFactory f
     }
   }
 
+  private void configureProxy(ElasticsearchSinkConnectorConfig config,
+                              HttpClientConfig.Builder builder) {
+    final String proxy =
+            config.getString(ElasticsearchSinkConnectorConfig.CONNECTION_PROXY_CONFIG);
+    if (proxy != null && !proxy.isEmpty()) {
+      try {
+        String protocol = proxy.split("://")[0];
+        String host = proxy.split("://")[1].split(":")[0];
+        int port = Integer.parseInt(proxy.split(":")[2]);
+
+        builder.proxy(new HttpHost(host, port, protocol));
```

Comment:
quick question, why are you using as well the authentication system for the proxy? some proxies I've been using require you some sort of it as well. what do you think?

Reply:
@purbon, not sure what you mean?

Reply:
I think @purbon refers to the `user:pass` part that is common to proxies. Can be supplied within the proxy url, but as I mention in my example below, I think it's better if it's supported as a separate config property like we do in the S3 connector.